src/Sema.zig

```diff
@@ -12417,7 +12417,55 @@ fn zirIntToPtr(sema: *Sema, block: *Block, inst: Zir.Inst.Index) CompileError!Ai
 fn zirErrSetCast(sema: *Sema, block: *Block, inst: Zir.Inst.Index) CompileError!Air.Inst.Ref {
     const inst_data = sema.code.instructions.items(.data)[inst].pl_node;
     const src = inst_data.src();
-    return sema.fail(block, src, "TODO: Sema.zirErrSetCast", .{});
+    const dest_ty_src: LazySrcLoc = .{ .node_offset_builtin_call_arg0 = inst_data.src_node };
+    const operand_src: LazySrcLoc = .{ .node_offset_builtin_call_arg1 = inst_data.src_node };
+    const extra = sema.code.extraData(Zir.Inst.Bin, inst_data.payload_index).data;
+
+    const dest_ty = try sema.resolveType(block, dest_ty_src, extra.lhs);
+    if (dest_ty.zigTypeTag() != .ErrorSet) {
```

Comment:
There's a pattern here we can follow to slightly reduce duplication, see some of these other functions for example:

```
fn checkIntType(
fn checkPtrType(
fn checkVectorElemType(
fn checkFloatType(
fn checkNumericType(
fn checkAtomicOperandType(
fn checkIntOrVector(
```

Looks like operand_ty below wants the exact same check.